### ISSUE TITLE
Howl may not startup with very large files in session

### ISSUE THREAD

**Issue by shalabhc** [howl-editor/howl#187]
I opened a log file in Howl and it was added to the session. Overnight the log file grew and the next day I got an error when trying to start Howl:

```
PANIC: unprotected error in call to Lua API (not enough memory)
```

I traced this down to the large file, which was now 700MB. So it's reasonable that Howl didn't handle it (at least not yet :wink: ).

Maybe Howl should check the file size and refuse to open (or reopen) files above a certain size. 

I was able to open a 35MB file without issues so maybe we set the limit to 30MB?

**Comment by nilnor** [comment]
Yes, this would be a good idea, or at least prompt the user for files over a certain limit. 30MB seems as good a limit as any for a hard cut-off, but if we go the prompting route this could be lower.

**Comment by tastyminerals** [comment]
Hi guys, just wanted to give some feedback about large files. I have an 48MB xml file with over 1,300,000 lines and when I jump through it **Howl** complains that there is either not enough memory or it just crashes. However, if I slowly go through it I can reach the last line without a problem.

**Comment by kirbyfan64** [comment]
My guess is that it tries to refresh the whole styling at once, but going through it slowly has it refresh in small intervals.

Maybe just have a cap on how far to refresh the styling at once?

**Comment by nilnor** [comment]
@kirbyfan64 Your guess is right. The reason it works in @tastyminerals latter case is that the buffer contents and styling is allocated using the FFI which allows for larger allocations, while lexing (using LPEG) is restricted to Lua's allocator.

It's not possible to just naively enforce a cap however, since we really need to style up to whatever is visible (hence the out-of-memory when jumping through it). So a cap would have to be smart, breaking up the styling into multiple lexing passes and intelligently determining where to resume it - but everything up until the visible area needs to be styled.

**Comment by nilnor** [comment]
Sorry for the very long silence here.

Howl development has been quiet on `master` for years, while work continued on the `next` branch — that branch has now been merged. Master has changed substantially since the 0.6 release: a full port to GTK 4, and a rewrite of the command line and interaction APIs, among a lot else.

We're closing out the old backlog, and feature requests are being closed for good rather than parked. Howl is developed at a pace that doesn't support an open wishlist, and leaving these sitting here indefinitely implies a roadmap that doesn't exist. That's not a comment on the merit of the idea — thank you for taking the time to write it up.

Please don't reopen this one. Bug reports against current master are still welcome.
